src/picongpu/include/particles/synchrotronPhotons/PhotonCreator.hpp

```diff
@@ -324,15 +324,16 @@ struct PhotonCreator
     template<typename Electron, typename Photon>
     DINLINE void operator()(Electron& electron, Photon& photon) const
     {
+        namespace parOp = PMacc::particles::operations;
         PMACC_AUTO(destPhoton,
-            PMacc::particles::operations::deselect<
+            parOp::deselect<
                 boost::mpl::vector<
                     multiMask,
                     momentum
                 >
             >(photon)
         );
-        PMacc::particles::operations::assign( destPhoton, electron );
+        parOp::assign( destPhoton, parOp::deselect<particleId>(electron) );
```

Comment:
all matching == all but multMask & momentum shall be copied BUT NOT particleID which will be filled with it's default value, which is a new ID